### DISCUSSION TITLE
Other version of  krita ai diffusion

### DISCUSSION THREAD

**Discussion by wilzamguerrero** [Acly/krita-ai-diffusion#1001] [General]
Hello, thank you for this tool, it is incredible, I am a designer and I have some basic programming knowledge, the truth is I would like to have another option for this tool using the native comfyui nodes, for example this tool does it but with photoshop https://github.com/zombieyang/sd-ppp, I would like to know If you have a basic code base where it allows me to send and receive images between krita and comfyui as it is in that script and if you can share the code to continue developing it, so far I have something basic like the panel in krita and the krita reception node at comfyui, any help is welcome, thank you very much for everything

**Comment by Acly** [comment]
There was a lot of discussion about something like this in #385

I see two ways to do it:
1. Launch execution in the Comfy web client and pull/send images from/to Krita (this is how the PS plugin you link works)
2. Launch execution in Krita client and pull the workflow (nodes) from the web UI

The plugin uses https://github.com/Acly/comfyui-tooling-nodes?tab=readme-ov-file#sending-and-receiving-images to send/receive images, but this only works for method 2. It's also the way I would prefer, because it allows easy parametrization from Krita for the workflows you design in web UI.

If you want to follow the PS plugin design you will have to write a server for the Krita side and have some kind of bidirectional flow.
